### ISSUE TITLE
Omnibar style control for keyboard navigation

### ISSUE THREAD

**Issue by jkantro** [decarbonization/m9r#22]
M9r is a simple music player that should be keyboard navigable. I recommend we approach this by an omnibar type experience given limited OOB support for keyboard navigation. In the future, Omnibar gives us a surface area to support search, typeahead, command based actions and more. It could potentially be extensible. This idea is inspired by notational velocity meets Raycast.

**Comment by decarbonization** [comment]
On a technical level, this can probably be done by asking all of the visible children of `ContentView` to donate actions when the omnibar is called up by the user. This would handily cover things like the side bar, play queue, playback controls, and whatever content the user happens to be browsing at this time. The pipeline for this could probably be built on top of a [`.task(priority:_:)`](https://developer.apple.com/documentation/swiftui/view/task(priority:_:)) listening for messages from an [`AsyncChannel`](https://github.com/apple/swift-async-algorithms/blob/main/Sources/AsyncAlgorithms/Channels/AsyncChannel.swift). SwiftUI tasks are only active so long as a view is visible, so this would avoid dead actions from lingering around. As far as the actual omnibar getting on screen, it can probably be done with another `Window` combined with the [`.plain`](https://developer.apple.com/documentation/swiftui/windowstyle/plain) window style.

**Comment by decarbonization** [comment]
I think it may make sense to support hierarchical actions. Think

- Select Playlist
  + Favorites
  + Soundtrack
  + ...
